### ISSUE TITLE
Storybook 6.1.0 Docs styling gone

### ISSUE THREAD

**Issue by simonerlandsen** [storybookjs/storybook#13170]
**Describe the bug**
Updated Storybook and addons from 6.0.22 to 6.1.0 and the styling disapeared from the docs pages.

**To Reproduce**

1. Upgrade Storybook and addons to 6.1.0
2. run `npm run storybook`
3. Go to a docs page and see that the styling for headers and code snippets and stuff are gone.

**Expected behavior**
Expected to see same styling on docs pages as before

**Screenshots**
Before:
![image](https://user-images.githubusercontent.com/8102223/99661614-6684af80-2a64-11eb-947a-90a4d3f994ab.png)

After:
![image](https://user-images.githubusercontent.com/8102223/99661669-77cdbc00-2a64-11eb-93de-d98c3366bc0a.png)

**System**
Environment Info:

  System:
    OS: Windows 10 10.0.18363
    CPU: (8) x64 Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
  Binaries:
    Node: 14.15.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.8 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 87.0.4280.66
    Edge: Spartan (44.18362.449.0)
  npmPackages:
    @storybook/addon-a11y: 6.1.0 => 6.1.0
    @storybook/addon-docs: 6.1.0 => 6.1.0
    @storybook/addon-essentials: 6.1.0 => 6.1.0
    @storybook/addon-knobs: 6.0.22 => 6.0.22
    @storybook/addons: 6.1.0 => 6.1.0
    @storybook/react: 6.1.0 => 6.1.0

**Additional context**
`npx sb@next info` lists Windows as operating system, but am running it in a linux node:14 container (have ran docker-compose up --build --force-recreate to ensure that the newest dependencies is installed on the image).

**Comment by shilman** [comment]
Ouch! Do you have a repro repo you can share? Def not seeing that in our projects..

**Comment by simonerlandsen** [comment]
I tried to reproduce by using `npx create-react-app my-app` (although I do not use creact-react-app in my other project) and `npx sb init` and then add an mdx file that I had in the other repo and the bug does not manifest.

What I am noticing though is that in my project with the bug, the header elements in the docs does not get the css classes that the newly made project does, and code snippets gets more transformed in the new project..

In the new project one code snippet compiles to
```html
<pre class="prismjs css-4zr3vl"><code class="language-javascript css-1maezg8"><span class="token keyword module">import</span><span class=""> </span><span class="token imports token punctuation">{</span><span class="token imports"> nameOfComponent </span><span class="token imports token punctuation">}</span><span class=""> </span><span class="token keyword module">from</span><span class=""> </span><span class="token string">"buflib"</span><span class="token punctuation">;</span></code></pre>
```

but the same snippet in my other project gets
```html
<pre><code class="language-javascript">import { nameOfComponent } from 'buflib';
</code></pre>
```
in the other.

I am adding some config to webpack in my main file. Can it have something to do with that?

My main file looks like this:

```javascript
const path = require('path');

const includes = [
  path.resolve(__dirname, '../stories'),
  path.resolve(__dirname, '../.storybook/styles'),
  path.resolve(__dirname, '../buflib'),
];

module.exports = {
  stories: ['../stories/**/*.stories.@(js|mdx)'],
  addons: [
    {
      name: '@storybook/addon-essentials',
      options: {
        actions: false,
      },
    },
    '@storybook/addon-docs',
    '@storybook/addon-knobs/register',
    '@storybook/addon-a11y',
  ],
  webpackFinal: async (config, { configType }) => {
    // `configType` has a value of 'DEVELOPMENT' or 'PRODUCTION'
    // You can change the configuration based on that.
    // 'PRODUCTION' is used when building the static version of storybook.

    // Make whatever fine-grained changes you need

    config.resolve.alias = { buflib: path.resolve(__dirname, '../buflib') };

    config.resolve.extensions.push(
      '.sass',
      '.scss',
      '.css',
      '.wasm',
      '.web.js',
      '.mjs',
      '.js',
      '.json',
      '.web.jsx',
      '.jsx',
    );

    config.module.rules.push({
      test: /\.(js|jsx|mjs)$/,
      include: includes,
      use: [
        {
          loader: 'babel-loader',
          options: {
            compact: true,
            presets: ['@babel/preset-env', '@babel/preset-react'],
            plugins: ['react-docgen'],
          },
        },
      ],
    });

    config.module.rules.push({
      test: /\.scss$/,
      use: [
        'style-loader',
        'css-loader?sourceMap=true',
        'sass-loader?sourceMap=true',
      ],
      include: includes,
    });

    // Return the altered config
    return config;
  },
};
```

**Comment by simonerlandsen** [comment]
And also, thank you so much for looking at my issue, and also for the amazing storybook project!

**Comment by simonerlandsen** [comment]
Okay, tried the old main.js in the newly created project and installed the same addons, but it does not fault like the other one. Tried to delete node_modules folder and package-lock and install again, but to no avail. Hm, I am at a loss. 

Maybe I should make new project and move all the stories and components into to that project. I am going to investigate a little bit more before trying that, and also give this issue some time and see if anyone can point me in the right direction.

**Comment by simonerlandsen** [comment]
Got it to work! I had the docs addon and the mdx-loader in my package.json because my Storybook instance failed without them in my project at an earlier version of Storybook. Uninstalled those and also removed docs from the main.js as it comes with essentials.

Tried to build again, but it failed due to missing mdx-loader, so obviously when I uninstalled the loader, npm also removed it for other dependencies that needed it, which I don't know if should work like that. Anyways, I deleted node_modules and package-lock again, installed and now it builds and the styles on the docs pages are back.

So the initial problem was probably that I had another version of the mdx-loader that what is used in the latest version of Storybook.

Thanks again for this the awesome Storybook project. It helps my team so much in everyday frontend development and testing.

**Comment by shilman** [comment]
@simonerlandsen so glad you were able to figure it out. i think there were some breaking changes in MDX despite no semver bump ... I can't remember what they were but I updated to the latest version at some point during 6.0 and had to make some changes. I guess if you had the old version installed it would cause problems.

Closing this now. Thanks for getting it on my radar!

**Comment by renadr** [comment]
I have exactly the same problem and I've tried a lot of things and I don't see anyone having the same problem yet. All of this since I moved from 6.0.22 to 6.1.9 (and also 6.1.10).

My `main.js` :

```
module.exports = {
  stories: ['../src/**/*.stories.mdx', '../src/**/stories.@(ts|js|tsx|mdx)'],
  addons: [
    '@storybook/preset-create-react-app',
    '@storybook/addon-essentials',
  ],
};
```

My storybook's `devDependencies` on `package.json`:

```
    "@storybook/addon-docs": "6.1.10",
    "@storybook/addon-essentials": "6.1.10",
    "@storybook/addon-knobs": "6.1.10",
    "@storybook/addon-options": "^5.3.21",
    "@storybook/preset-create-react-app": "^3.1.4",
    "@storybook/react": "6.1.10",
    "@storybook/source-loader": "6.1.10",
```

My MDX file : 

```
import { Meta } from '@storybook/addon-docs/blocks';

<Meta title="Introduction" />

# Blabla

## Blablabla

Lorem ipsum
```

So the html structure is good at the end but there is no class 'css-blabla'.

**Comment by shilman** [comment]
@renadr can you remove `"@storybook/addon-options": "^5.3.21"` and remove node_modules and reinstall?

**Comment by renadr** [comment]
Thank you for your response 😃. Actually I just noticed that it worked for a while locally but not once published (I thought I had a cache problem or whatever on the local). I finally noticed it was probably coming from the CI and I emptied the cache of the CI, and published it and it finally worked ! So... I had just lost my mind 🤯.

I'll try to post another message tomorrow to give a config example that led me to this bug (I think I have a branch where I already had this bug). I hope it will help to find the source of the problem.
